### ISSUE TITLE
Synchronizing UserSocialAuth doesn't work with duplicated GitHub usernames

### ISSUE THREAD

**Issue by chrismedrela** [swcarpentry/amy#891]
When we have two people with the same GitHub username and UserSocialAuth is already synchronized for the first person, we cannot synchronize for the another one. This is because one GitHub account can be associated with only one AMY account. This results in `IntegrityError` raised in `sync_usersocialauth` view.

What should we do? What about displaying an error message (I mean `messages.error`) with information who has the same GitHub account?

**Comment by gvwilson** [comment]
How can we have two people with the same GitHub username?

**Comment by chrismedrela** [comment]
Yeah, I forgot that we have uniqueness constraint on `Person.github`, but this error still may happen. Consider this scenario:

1) We set GitHub username of person A to `chrismedrela` and synchronize `UserSocialAuth`.
2) We then change the username to something else, but we don't update `UserSocialAuth` table. Person A can still login using `chrismedrela` GitHub account.
3) We set GitHub username of person B to `chrismedrela` and try to synchronize `UserSocialAuth`.

It turns out that we probably should clean `UserSocialAuth` every time we change `Person.github`. Any opinions?
